src/main/java/software/amazon/encryption/s3/internal/ContentMetadataStrategy.java

```diff
@@ -33,16 +34,15 @@ public abstract class ContentMetadataStrategy implements ContentMetadataEncoding
     public static final ContentMetadataDecodingStrategy INSTRUCTION_FILE = new ContentMetadataDecodingStrategy() {
 
         @Override
-        public ContentMetadata decodeMetadata(GetObjectRequest getObjectRequest, GetObjectResponse response) {
+        public ContentMetadata decodeMetadata(GetObjectRequest getObjectRequest, GetObjectResponse response, S3AsyncClient s3AsyncClient) {
             GetObjectRequest instructionGetObjectRequest = GetObjectRequest.builder()
                     .bucket(getObjectRequest.bucket())
                     .key(getObjectRequest.key() + INSTRUCTION_FILE_SUFFIX)
                     .build();
 
-            S3Client s3Client = S3Client.create();
-            ResponseInputStream<GetObjectResponse> instruction;
+            ResponseBytes<GetObjectResponse> instruction;
```

Comment:
Why use `ResponseBytes` instead of InputStream? Instruction file is small, but we might as well keep using the stream approach. 

  ```suggestion
            try (ResponseInputStream<GetObjectResponse> instruction = s3AsyncClient.getObject(instructionGetObjectRequest, AsyncResponseTransformer.toBlockingInputStream()).join()){
                Map<String, String> metadata = new HashMap<>();
                JsonNodeParser parser = JsonNodeParser.create();
                JsonNode objectNode = parser.parse(instruction);
                for (Map.Entry<String, JsonNode> entry : objectNode.asObject().entrySet()) {
                    metadata.put(entry.getKey(), entry.getValue().asString());
                }
                return ContentMetadataStrategy.readFromMap(metadata, response);
            } catch (IOException ioException) {
                throw new S3EncryptionClientException("IOException while reading instruction file!", ioException);
            } catch (NoSuchKeyException exception) {
                // Most likely, the customer is attempting to decrypt an object
                // which is not encrypted with the S3 EC.
                throw new S3EncryptionClientException("Instruction file not found! Please ensure the object you are" +
                        " attempting to decrypt has been encrypted using the S3 Encryption Client.", exception);
            }
```

Reply:
Agreed! Will update the PR